code/datums/storage/storage.dm

```diff
@@ -952,32 +959,31 @@ GLOBAL_LIST_EMPTY(cached_storage_typecaches)
 			resolve_parent.balloon_alert(to_show, "locked!")
 		return FALSE
 
-	if(!quickdraw || to_show.get_active_held_item())
-		if(display_contents)
-			show_contents(to_show)
-
-		if(animated)
-			animate_parent()
-
-		if(rustle_sound)
-			playsound(resolve_parent, SFX_RUSTLE, 50, TRUE, -5)
-
-		return TRUE
-
-	var/obj/item/to_remove = locate() in resolve_location
+	// If we're quickdrawing boys
+	if(quickdraw && !to_show.get_active_held_item())
+		var/obj/item/to_remove = locate() in resolve_location
+		if(!to_remove)
+			return TRUE
 
-	if(!to_remove)
+		attempt_remove(to_remove)
+		INVOKE_ASYNC(src, PROC_REF(put_in_hands_async), to_show, to_remove)
+		if(!silent)
+			to_show.visible_message(span_warning("[to_show] draws [to_remove] from [resolve_parent]!"), span_notice("You draw [to_remove] from [resolve_parent]."))
```

Comment:
```suggestion
			to_show.visible_message(
				span_warning("[to_show] draws [to_remove] from [resolve_parent]!"),
				span_notice("You draw [to_remove] from [resolve_parent]."),
			)
```

multiline to make it a bit nicer since it's a bit long imo